src/runtime.ts

```diff
@@ -93,6 +93,31 @@ function mapInternal<A extends ValueType, T extends ValueType>(
   return value;
 }
 
+export function filterDeep<A extends ValueType, T extends ValueType>(
```

Comment:
Filter to me sounds like a structure preserving operation. Could we instead do this as `forEach` and then where we need to do this say `runtime.forEach(predicate, found.push)`

Reply:
That's a good point actually. There's a lodash extension library with `filterDeep` that indeed preserves the structure.
https://deepdash.io/#filterdeep

The name aside, this functionality is exactly the one that I'd need. That's why I'd rather have exactly this functionality and call it e.g. `getAll`, than add `forEach` and using it to populate an array.
What do you think?

Or is your idea that the `forEach` would be useful for some other cases as well?

Reply:
`getAll` is fine